pkg/asset/machines/openstack/machines.go

```diff
@@ -46,6 +46,11 @@ func Machines(clusterID string, config *types.InstallConfig, pool *types.Machine
 		return nil, err
 	}
 
+	volumeAZs := []string{""}
+	if mpool.RootVolume != nil && mpool.RootVolume.Zones != nil {
```

Comment:
I think that you want to use the zones from the root volume only if the slice is non-empty. Otherwise, if it is an empty non-nil slice, you will get a panic later when you try to use the length of `volumeAZs` as a modulus later. I think, following through how the pool gets defined and set, that the zones will not be an empty non-nil slice, but it would still be better to have the correct check to be resilient to future changes with how the pool is set up.
```suggestion
	if mpool.RootVolume != nil && len(mpool.RootVolume.Zones) != 0 {
```